lib/logstash/filters/aggregate.rb

```diff
@@ -317,13 +348,31 @@ def filter(event)
       # retrieve the current aggregate map
       aggregate_maps_element = @@aggregate_maps[task_id]
       
+
       # create aggregate map, if it doesn't exist
       if (aggregate_maps_element.nil?)
         return if @map_action == "update"
         # create new event from previous map, if @push_previous_map_as_event is enabled
         if (@push_previous_map_as_event and !@@aggregate_maps.empty?)
           previous_map = @@aggregate_maps.shift[1].map
           event_to_yield = LogStash::Event.new(previous_map)
+
+          if @timeout_task_id_field
+            event_to_yield[@timeout_task_id_field] = task_id
+          end
+
+          # Call code block if available
+          if @timeout_code
+            begin
+              @timeout_codeblock.call(event)
+              events_to_flush << event
+              noError = true
+            rescue => exception
+              @logger.error("Aggregate exception occurred. Error: #{exception} ; Code: #{@timeout_code} ; Map: #{map} ; EventData: #{event.instance_variable_get('@data')}")
```

Comment:
To clearly separate "code" from "timeout_code" errors, could you :
- replace "Code:" by "TimeoutCode:"
- replace "Map:" by "PreviousMap:"
- replace "#{map}" by "#{previous_map}"
- replace "EventData:" by "TimeoutEventData"
- replace "event." by "event_to_yield."